### PR TITLE
Update Tarantool versions in test matrix

### PR DESCRIPTION
**Pull request by ligurio**
Tarantool 2.8.1 [1] and Tarantool 2.9 are available [2], let's include
it to test matrix. Test
simple_operations[memtx].test_intermediate_nullable_fields_update has
failed on 2.8 [3] and to crash looks similar to a bug described in [4].
Testcase skipped on 2.8 to workaround it before next release.

Tarantool 2.4-6 has reached EOL, so these version has been removed from
test matrix.

1. https://github.com/tarantool/tarantool/releases/tag/2.8.1
2. https://github.com/tarantool/tarantool/releases/tag/2.9.0
3. https://github.com/tarantool/crud/pull/195/checks?check_run_id=3320469716
4. https://github.com/tarantool/tarantool/issues/6069

What has been done? Why? What problem is being solved?

I didn't forget about

- [ ] Tests
- [ ] Changelog
- [ ] Documentation

Closes #???

### CODE REVIEW

**Review by olegrok — APPROVED** [review]

**Comment by ligurio** [comment]
@Totktonada, could you mark testing for Tarantool 2.{4,5,6} as not required in repository settings to make CI green?

File: test/integration/simple_operations_test.lua [hunk 1]
```diff
@@ -394,6 +394,8 @@ pgroup:add('test_upsert', function(g)
 end)
 
 pgroup:add('test_intermediate_nullable_fields_update', function(g)
+    -- Blocked by https://github.com/tarantool/tarantool/issues/6069
+    t.skip_if(_TARANTOOL > "2.8")
```
Comment (Totktonada) [inline test/integration/simple_operations_test.lua:398]: The condition looks strange.

```sh
$ (needle='tuple: fix crash in reversed nullable json update'; for branch in 1.7 1.9 1.10 2.1 2.2 2.3 2.4 2.5 2.6 2.7 2.8 master; do printf "${branch}\t"; commit_description=$(git log --oneline "origin/${branch}" | grep -F "${needle}"); [ -n "${commit_description}" ] && echo "$(git describe --long --always $(echo "${commit_description}" | cut -d" " -f1)) $(echo "${commit_description}" | cut -d" " -f2-)" || echo "(not found)"; done)
1.7	(not found)
1.9	(not found)
1.10	(not found)
2.1	(not found)
2.2	(not found)
2.3	(not found)
2.4	(not found)
2.5	(not found)
2.6	(not found)
2.7	(not found)
2.8	2.8.1-13-g6e0c615e6 tuple: fix crash in reversed nullable json update
master	2.9.0-26-g75b2ccce8 tuple: fix crash in reversed nullable json update
```

It seems it is the regression from the following commit:

```sh
$ (needle='update: allow update absent nullable fields'; for branch in 1.7 1.9 1.10 2.1 2.2 2.3 2.4 2.5 2.6 2.7 2.8 master; do printf "${branch}\t"; commit_description=$(git log --oneline "origin/${branch}" | grep -F "${needle}"); [ -n "${commit_description}" ] && echo "$(git describe --long --always $(echo "${commit_description}" | cut -d" " -f1)) $(echo "${commit_description}" | cut -d" " -f2-)" || echo "(not found)"; done)
1.7	(not found)
1.9	(not found)
1.10	(not found)
2.1	(not found)
2.2	(not found)
2.3	(not found)
2.4	(not found)
2.5	(not found)
2.6	(not found)
2.7	(not found)
2.8	2.8.0-223-g2bb373b9f update: allow update absent nullable fields
master	2.8.0-223-g2bb373b9f update: allow update absent nullable fields
```

So we should skip the test for versions in the following ranges: [2.8.0-223, 2.8.1-13), [2.9.0-0, 2.9.0-26).

**Comment by Totktonada** [comment]
> @Totktonada, could you mark testing for Tarantool 2.{4,5,6} as not required in repository settings to make CI green?

Done.

File: test/integration/simple_operations_test.lua (hunk 1 above)
Comment (olegrok) [reply test/integration/simple_operations_test.lua:398]: Yes, currently we could drop this check (it was introduced before 2.8.2 was released)

File: .github/workflows/test_on_push.yaml [hunk 2]
```diff
@@ -11,7 +11,7 @@ jobs:
       github.event_name == 'pull_request' && github.event.pull_request.head.repo.full_name != github.repository
     strategy:
       matrix:
-        tarantool-version: ["1.10", "2.2", "2.3", "2.4", "2.5", "2.6", "2.7"]
+        tarantool-version: ["1.10", "2.2", "2.3", "2.7", "2.8", "2.9"]
```
Comment (Totktonada) [inline .github/workflows/test_on_push.yaml:14]: 2.9 was never released. There was a 'technical release' 2.9.0, which is equivalent to 2.8.1. All development goes into 2.10. Now there is one pre-release: 2.10.0-beta1. See https://github.com/tarantool/setup-tarantool/issues/19.

File: CHANGELOG.md [hunk 3]
```diff
@@ -9,6 +9,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ### Changed
 
+* Tarantool versions 2.{4,5,6} have reached EOL and removed from regression
+  testing.
```
Comment (olegrok) [inline CHANGELOG.md:13]: It's worth to mention that min/max commands won't work for old 1.10 versions anymore.

Comment (olegrok) [reply CHANGELOG.md:13]: And it's a bit inconsistent with select that still works for such versions.

File: .github/workflows/test_on_push.yaml (hunk 2 above)
Comment (ligurio) [reply .github/workflows/test_on_push.yaml]: 2.9 removed

File: test/integration/simple_operations_test.lua (hunk 1 above)
Comment (ligurio) [reply test/integration/simple_operations_test.lua]: > Yes, currently we could drop this check (it was introduced before 2.8.2 was released)

done

File: CHANGELOG.md [hunk 4]
```diff
@@ -9,6 +9,12 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ### Changed
 
+* Remove support of Tarantool versions that have no builtin tuple-keydef
+  module.
+* Tarantool versions 2.{4,5,6} have reached EOL and removed from regression
+  testing. It means that methods `crud.min()` and `crud.max` won't work for old
+  1.10 versions anymore and it's a bit inconsistent with select that still
```
Comment (olegrok) [inline CHANGELOG.md:16]: > it's a bit inconsistent with select that still
  works for such versions

It was my opinion :) You shouldn't copy-paste it as is) It looks strange.

Comment (ligurio) [reply CHANGELOG.md]: updated

File: CHANGELOG.md (hunk 3 above)
Comment (Totktonada) [reply CHANGELOG.md]: How one is relevant to another?

**Pull request merged**